framework/include/base/MooseApp.h

```diff
@@ -312,26 +301,12 @@ class MooseApp :
 
   /**
    * Store a map of outputter names and file numbers
-   * The MultiApp system requires this to get the file numbering to propagate down through the Multiapps.
-   * @param numbers Map of outputter names and file numbers
-   *
-   * @see MultiApp TransientMultiApp OutputWarehouse
-   */
-  void setOutputFileNumbers(std::map<std::string, unsigned int> numbers) { _output_file_numbers = numbers; }
-
-  /**
-   * Store a map of outputter names and file numbers
    * The MultiApp system requires this to get the file numbering to propogate down through the
    * multiapps.
    *
    * @see MultiApp TransientMultiApp
    */
-  std::map<std::string, unsigned int> & getOutputFileNumbers() { return _output_file_numbers; }
-
-  /**
-   * Return true if the output position has been set
-   */
-  bool hasOutputWarehouse(){ return _output_position_set; }
+  std::map<std::string, unsigned int> getOutputFileNumbers();
```

Comment:
Ok - sounds reasonable.  Whenever I see a an object like that getting returned by value there's just a flag that goes up.